Modify the buildsystem to work with the legacy version of samtools,
which can coexist with the most recent version of samtools.
See also: https://bugs.gentoo.org/show_bug.cgi?id=577854

--- cufflinks-2.2.1/ax_bam.m4
+++ cufflinks-2.2.1/ax_bam.m4
@@ -102,7 +102,7 @@
 
 	AC_LANG_PUSH(C++)
      	AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
-	@%:@include <bam/bam.h>
+	@%:@include <bam-0.1-legacy/bam.h>
 	]], [[
 	]])],[
         AC_MSG_RESULT(yes)
@@ -171,7 +171,7 @@
 
 		AC_LANG_PUSH(C++)
 	     	AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
-		@%:@include <bam/version.hpp>
+		@%:@include <bam-0.1-legacy/version.hpp>
 		]], [[
 		]])],[
         	AC_MSG_RESULT(yes)
@@ -189,7 +189,7 @@
 			AC_MSG_NOTICE([Your bam libraries seem too old (version $_version).])
 		fi
 	else
-		BAM_LIB="-lbam"
+		BAM_LIB="-lbam-0.1-legacy"
 		AC_SUBST(BAM_CPPFLAGS)
 		AC_SUBST(BAM_LDFLAGS)
 		AC_SUBST(BAM_LIB)
--- cufflinks-2.2.1/src/hits.h
+++ cufflinks-2.2.1/src/hits.h
@@ -18,7 +18,7 @@
 
 #include <boost/shared_ptr.hpp>
 
-#include <bam/sam.h>
+#include <bam-0.1-legacy/sam.h>
 
 #include "common.h"
 #include "multireads.h"
